common/src/main/java/tachyon/worker/WorkerClient.java

```diff
@@ -87,8 +87,10 @@ public WorkerClient(NetAddress workerNetAddress, ExecutorService executorService
     mSessionId = sessionId;
     mIsLocal = isLocal;
     mClientMetrics = Preconditions.checkNotNull(clientMetrics);
+    mHeartbeatExecutor = new WorkerClientHeartbeatExecutor(this);
```

Comment:
Moved initialization on the constructor. No need to recreate it at every connect.